src/libraries/Microsoft.Extensions.Logging.Console/tests/Microsoft.Extensions.Logging.Console.Tests/ConsoleLoggerProcessorTests.cs

```diff
@@ -0,0 +1,176 @@
+        }
+
+        [ConditionalTheory(typeof(PlatformDetection), nameof(PlatformDetection.IsThreadingSupported))]
+        [InlineData(-1)]
+        [InlineData(0)]
+        public static void MaxQueueLength_SetInvalid_Throws(int invalidMaxQueueLength)
+        {
+            // Arrange
+            var sink = new ConsoleSink();
+            var console = new TestConsole(sink);
+            var processor = new ConsoleLoggerProcessor(console, null!, ConsoleLoggerBufferFullMode.Wait, 1024);
+
+            // Act & Assert
+            Assert.Throws<ArgumentOutOfRangeException>(() => processor.MaxQueueLength = invalidMaxQueueLength);
+        }
+
+        [ConditionalFact(typeof(PlatformDetection), nameof(PlatformDetection.IsThreadingSupported))]
+        public static void FullMode_SetInvalid_Throws()
+        {
+            // Arrange
+            var sink = new ConsoleSink();
+            var console = new TestConsole(sink);
+            var processor = new ConsoleLoggerProcessor(console, null!, ConsoleLoggerBufferFullMode.Wait, 1024);
+
+            // Act & Assert
+            Assert.Throws<ArgumentOutOfRangeException>(() => processor.FullMode = (ConsoleLoggerBufferFullMode)10);
+        }
+
+        [OuterLoop]
+        [ConditionalFact(typeof(PlatformDetection), nameof(PlatformDetection.IsThreadingSupported))]
+        [InlineData(true)]
+        [InlineData(false)]
+        public void CheckForNotificationWhenQueueIsFull(bool okToDrop)
+        {
+            // Arrange
+            var sink = new ConsoleSink();
+            var console = new TestConsole(sink);
+            var errorConsole = new TimesWriteCalledConsole();
+            string queueName = nameof(CheckForNotificationWhenQueueIsFull) + (okToDrop ? "InDropWriteMode" : "InWaitMode");
+            var processor = new ConsoleLoggerProcessor(console, errorConsole, ConsoleLoggerBufferFullMode.Wait, 1024);
+            var formatter = new SimpleConsoleFormatter(new TestFormatterOptionsMonitor<SimpleConsoleFormatterOptions>(
+                new SimpleConsoleFormatterOptions()));
+
+            var logger = new ConsoleLogger(_loggerName, processor, formatter, null, new ConsoleLoggerOptions());
+            Assert.Null(logger.Options.FormatterName);
+            UpdateFormatterOptions(logger.Formatter, logger.Options);
+            string messageTemplate = string.Join(", ", Enumerable.Range(1, 100).Select(x => "{A" + x + "}"));
+            object[] messageParams = Enumerable.Range(1, 100).Select(x => (object)x).ToArray();
+
+            // Act
+            processor.MaxQueueLength = 1;
+            processor.FullMode = okToDrop ? ConsoleLoggerBufferFullMode.DropWrite : ConsoleLoggerBufferFullMode.Wait;
+            for (int i = 0; i < 20000; i++)
+            {
+                logger.LogInformation(messageTemplate, messageParams);
+            }
+
+            // Assert
+            if (okToDrop)
+            {
+                Assert.True(errorConsole.TimesWriteCalled > 0);
```

Comment:
If the thread dequeing messages runs fast enough then nothing will be dropped and the test will fail. To make the test deterministic regardless of dequeue thread speed you would need to block at some point in the test console or console sink.